massa-pool-worker/src/worker.rs

```diff
@@ -99,28 +99,31 @@ impl OperationPoolThread {
 
     /// Run the thread.
     fn run(self, config: PoolConfig) {
-        let mut next_refresh = Instant::now();
+        let mut start_time = Instant::now();
+        let tick = config.operation_pool_refresh_interval.to_duration();
         loop {
-            match self.receiver.recv_deadline(next_refresh) {
-                Err(RecvTimeoutError::Disconnected) | Ok(Command::Stop) => break,
-                Ok(Command::AddItems(operations)) => {
-                    self.operation_pool.write().add_operations(operations)
-                }
-                Ok(Command::NotifyFinalCsPeriods(final_cs_periods)) => self
-                    .operation_pool
-                    .write()
-                    .notify_final_cs_periods(&final_cs_periods),
-                Ok(_) => {
-                    warn!("OperationPoolThread received an unexpected command");
-                    continue;
-                }
-                Err(RecvTimeoutError::Timeout) => {}
-            };
-            if next_refresh <= Instant::now() {
+            let now = Instant::now();
+            let duration = (start_time + tick).saturating_duration_since(Instant::now());
+            if !duration.is_zero() {
+                let duration = start_time + tick - now;
```

Comment:
Why do you recompute `duration` here? Can't you just use its value?

Reply:
because i'm dumb :). Fixed :)